libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -464,120 +465,335 @@ def extract_images_from_page(self, page: pypdf._page.PageObject) -> str:
 
 
 class PDFMinerParser(BaseBlobParser):
-    """Parse `PDF` using `PDFMiner`."""
+    """Parse a blob from a PDF using `pdfminer.six` library.
+
+    This class provides methods to parse a blob from a PDF document, supporting various
+    configurations such as handling password-protected PDFs, extracting images, and
+    defining extraction mode.
+    It integrates the 'pdfminer.six' library for PDF processing and offers synchronous
+    blob parsing.
+
+    Examples:
+        Setup:
+
+        .. code-block:: bash
+
+            pip install -U langchain-community pdfminer.six pillow
+
+        Load a blob from a PDF file:
+
+        .. code-block:: python
+
+            from langchain_core.documents.base import Blob
+
+            blob = Blob.from_path("./example_data/layout-parser-paper.pdf")
+
+        Instantiate the parser:
 
-    def __init__(self, extract_images: bool = False, *, concatenate_pages: bool = True):
+        .. code-block:: python
+
+            from langchain_community.document_loaders.parsers import PDFMinerParser
+
+            parser = PDFMinerParser(
+                # password = None,
+                mode = "single",
+                pages_delimiter = "\n\f",
+                # extract_images = True,
+                # images_to_text = convert_images_to_text_with_tesseract(),
+            )
+
+        Lazily parse the blob:
+
+        .. code-block:: python
+
+            docs = []
+            docs_lazy = parser.lazy_parse(blob)
+
+            for doc in docs_lazy:
+                docs.append(doc)
+            print(docs[0].page_content[:100])
+            print(docs[0].metadata)
+    """
+
+    _warn_concatenate_pages = False
+
+    def __init__(
+        self,
+        extract_images: bool = False,
+        *,
+        password: Optional[str] = None,
+        mode: Literal["single", "page"] = "single",
+        pages_delimiter: str = _DEFAULT_PAGES_DELIMITER,
+        images_parser: Optional[BaseImageBlobParser] = None,
+        images_inner_format: Literal["text", "markdown-img", "html-img"] = "text",
+        concatenate_pages: Optional[bool] = None,
```

Comment:
This is a breaking change w/ respect due to the change in the default. 

Could we restore to how it was, update unit tests w/ an exception for this class (i.e., specifically pass concatenate_pages=False)

---

For making breaking changes in the API, let's not do this as part of the large PRs since they're hard to catch, and we'll want to document them properly / figure out how to roll them out.

Reply:
It's OK if the content of the Document changes due to improvements (e.g., the content is extracted in a format that better matches the PDF). We just don't want to cause silent failures in user code; e.g., 

user assumed it would be a single page and they're code does something like extract only the first document emitted

Reply:
@eyurtsev 
What do you think?

Reply:
Ah got it. OK looks fine!